src/libraries/System.Private.CoreLib/src/System/String.cs

```diff
@@ -45,6 +45,7 @@ public sealed partial class String : IComparable, IEnumerable, IConvertible, IEn
          */
 
         [MethodImpl(MethodImplOptions.InternalCall)]
+        [PreserveDependency("Ctor(System.Char[])", "System.String")]
```

Comment:
It's interesting that these are necessary.  For coreclr, the build system is currently generating an illink xml file containing everything the runtime references, which I expect includes the Ctor methods.  I assume then that this is necessary when building the corelib targeting mono?  Personally I'd prefer this approach wherever possible instead of having the build generated illink xml file.

Reply:
>  Personally I'd prefer this approach wherever possible instead of having the build generated illink xml file.

I agree that this is better, but I do not expect we will be able to get rid of the build generated illink xml file for CoreCLR anytime soon. It would mean translating the 1000+ lines of `vm\mscorlib.h` into PreserveDependency attributes and keeping the two in sync.

Reply:
In mono, we preserve them using a linker descriptor:
```
<type fullname="System.String" preserve="fields">
             <method name="Ctor"/>
</type>
```
But that is causes all of them to be preserved even if the corresponding ctor is not preserved.